firedantic/tests/tests_async/test_indexes.py

```diff
@@ -0,0 +1,156 @@
+from unittest.mock import AsyncMock
+
+from google.cloud.firestore_admin_v1 import ListIndexesResponse
+
+from firedantic import (
+    AsyncModel,
+    async_set_up_composite_indexes,
+    async_set_up_composite_indexes_and_ttl_policies,
+    collection_group_index,
+    collection_index,
+)
+from firedantic.tests.tests_async.conftest import MockListIndexOperation
+
+import pytest  # noqa isort: skip
+
+
+class BaseModelWithIndexes(AsyncModel):
+    __collection__ = "modelWithIndexes"
+
+    name: str
+    status: int
+    age: int
+
+
+@pytest.mark.asyncio
+async def test_set_up_composite_index(mock_admin_client):
+    class ModelWithIndexes(BaseModelWithIndexes):
+        __composite_indexes__ = (
+            collection_index(("name", "ASCENDING"), ("age", "DESCENDING")),
+        )
+
+    mock_admin_client.create_index.reset_mock()
```

Comment:
I think the mocks should be clean by default and test cases should not need to start with resetting a mock. For more details see https://github.com/ioxiocom/firedantic/pull/64/files#r1505817500